controlplane/kubeadm/controllers/controller.go

```diff
@@ -415,6 +423,10 @@ func (r *KubeadmControlPlaneReconciler) ClusterToKubeadmControlPlane(o handler.M
 // It removes any etcd members that do not have a corresponding node.
 // Also, as a final step, checks if there is any machines that is being deleted.
 func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, controlPlane *internal.ControlPlane) error {
+	if controlPlane.Machines.Len() == 0 {
```

Comment:
This check acts on machine objects, while `TargetClusterControlPlaneIsHealthy` and `TargetClusterEtcdIsHealthy` are acting on the actual list of nodes. 
However, I don't see cases when controlPlane.Machines.Len() == 0 and len(nodes) != 0 so this should be fine, right?

Reply:
Hmm, I'm not sure I follow. Are you concerned we might be skipping this too much or not enough because of this new check?

Reply:
just thinking loud if we are missing some case using machines vs nodes